control_apps/src/control_mngr.rs

```diff
@@ -145,15 +147,24 @@ impl ControlMngr {
     pub fn update(
         &mut self,
         altitude: Measurement<f32>,     // instantaneous altitude in meters
-        ascent_rate: Measurement<f32>,  // instantaneous ascent rate in meters per second
+        ascent_rate: Measurement<f32>,  // instantaneous ascent rate in m/s
         ballast_mass: Measurement<f32>, // ballast mass remining in kg
     ) -> ControlCommand {
-        self.abort_if_out_of_ballast(ballast_mass.value);
-        let error = altitude.value - self.target_altitude;
+        // log status information
         debug!(
-            "Altitude: {:} m ({:} m error) | State: {:} | {:} kg ballast left",
-            altitude.value, error, self.state, ballast_mass.value
+            "[{:}] Altitude {:} m ({:}), Ballast Remaining {:} kg ({:})",
+            self.state,
+            altitude.value, altitude.timestamp.elapsed().as_secs(),
+            ballast_mass.value, ballast_mass.timestamp.elapsed().as_secs()
         );
+
+        // abort if there's no ballast left, doesn't matter if telem is stale
+        self.abort_if_out_of_ballast(ballast_mass.value);
```

Comment:
commenting here because I can't comment on something thats not a dif??

It doesn't seem possible to recover from this.  Do we want some way to get out of the safe state after reading in 1 message with empty ballast? 

Also do we want to stop using the balloon valve even though the ballast is empty?

Reply:
Since dropping ballast is the only way for us to make the balloon rise after the initial ascent, I'm being super conservative here.

1. I think the answer is yes, even if there is a blip indicating that there _might_ not be any ballast left, we should abort because it would be basically mission ending if it is true and we continue to bleed gas. How likely do you think this is to occur?
2. Yes, if there is no more ballast we never want to release gas because there's no way to undo it. It's not hard to think of a reason this wouldn't be true in the ideal case, but I don't trust my controller enough.